fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/catalog/FlinkCatalog.java

```diff
@@ -359,15 +361,53 @@ public void alterTable(ObjectPath objectPath, CatalogBaseTable catalogBaseTable,
     @Override
     public List<CatalogPartitionSpec> listPartitions(ObjectPath objectPath)
             throws TableNotExistException, TableNotPartitionedException, CatalogException {
-        // TODO: use admin.listPartitionInfos()
-        return Collections.emptyList();
+
+        String tableName = objectPath.getObjectName();
+        TablePath tablePath;
+        try {
+            if (tableName.contains(LAKE_TABLE_SPLITTER)) {
+                tablePath =
+                        TablePath.of(
```

Comment:
```java
TablePath.of(
                                            objectPath.getDatabaseName(),
                                            tableName.split("\\" + LAKE_TABLE_SPLITTER)[0])))
```
has occured several time in catalog, maybe we can put into a method?

Reply:
I will modify it in the another PR  with another optimization.